src/app/components/Upload/Upload.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { uploadHighlight } from '../../utils/highlights';
+
+type UploadProps = {
+  src: string;
+};
+
+function Upload({ src }: UploadProps): JSX.Element {
+  const [url, setUrl] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [total, setTotal] = useState(0);
+  const [loaded, setLoaded] = useState(0);
+
+  async function handleUpload() {
+    try {
+      const url = await uploadHighlight({
+        src,
+        onProgress: ({ loaded, total }) => {
+          setTotal(total);
+          setLoaded(loaded);
+        },
+      });
+      setUrl(url);
+    } catch (error) {
+      console.error(error);
+      setErrorMessage(error.message);
+    }
+  }
+
+  return (
+    <>
+      <p>{src}</p>
+      <button onClick={handleUpload}>Upload</button>
+      <a href={url} target="_blank">
+        {url}
+      </a>
+      <label>
+        Upload Progress{' '}
+        <progress max={total} value={loaded}>
+          {(loaded / total) * 100}%
```

Comment:
```suggestion
          {(loaded * 100) / total}%
```

Because 100% = total, how much % ist loaded, Triangle calculations:

100 = total
? = loaded

;-)